platform_plugin_turnitin/turnitin_client/handlers.py

```diff
@@ -0,0 +1,280 @@
+import json
+from typing import Dict, Optional
+
+import requests
+
+from django.conf import settings
+
+
+def turnitin_api_handler(
+    request_method: str,
+    url_prefix: str = "",
+    data: Optional[Dict] = None,
+    is_upload: bool = False,
+    uploaded_file=None,
+):
+    """
+    Handles API requests to the Turnitin service.
+
+    Parameters:
+    - request_method (str): The HTTP method (e.g., 'GET', 'POST', 'PUT', 'PATCH', 'DELETE').
+    - data (dict): The payload to be sent in the request. Use None for methods that don't require a payload.
+    - url_prefix (str): The endpoint suffix for the API URL.
+
+    Returns:
+    - Response: A requests.Response object containing the server's response to the request.
+
+    Raises:
+    - ValueError: If an unsupported request method is provided.
+    """
+
+    # Configuration variables
+    TII_API_URL = getattr(settings, "TURNITIN_TII_API_URL", None)
+    TCA_INTEGRATION_FAMILY = getattr(settings, "TURNITIN_TCA_INTEGRATION_FAMILY", None)
+    TCA_INTEGRATION_VERSION = getattr(settings, "TURNITIN_TCA_INTEGRATION_VERSION", None)
+    TCA_API_KEY = getattr(settings, "TURNITIN_TCA_API_KEY", None)
+
+    # Headers configuration
+    headers = {
+        "X-Turnitin-Integration-Name": TCA_INTEGRATION_FAMILY,
+        "X-Turnitin-Integration-Version": TCA_INTEGRATION_VERSION,
+        "Authorization": f"Bearer {TCA_API_KEY}",
+    }
+
+    # Add Content-Type for methods that typically send JSON payload
+    if request_method.lower() in ["post", "put", "patch"]:
+        headers["Content-Type"] = "application/json"
+
+    if is_upload:
+        headers["Content-Type"] = "binary/octet-stream"
+        headers["Content-Disposition"] = f'inline; filename="{uploaded_file.name}"'
+        response = requests.put(
+            f"{TII_API_URL}/api/v1/{url_prefix}", headers=headers, data=uploaded_file
+        )
+        return response
+
+    # Mapping of methods to corresponding functions in the requests library
+    method_map = {
+        "get": requests.get,
+        "post": requests.post,
+        "put": requests.put,
+        "delete": requests.delete,
+        "patch": requests.patch,
+    }
```

Comment:
Instead of having a method_map, you could pass directly the method:
`turnitin_api_handler(requests.get, ...)`

Reply:
Hmm... I believe it would be better to create the method_map within a separate function. This way, we can modify the method conditionals based on a list of strings, rather than a list of methods. This approach:
 `if request_method.lower() in ["post", "put", "patch"]`  
is more elegant than this:
 `if request_method in [requests.patch, request.post,....] `
and certainly cleaner than a series of:
`if request_method == requests.post: 
    ...
elif ... : 
    ...
else:
    ...
`